boltdb/bucket.go

```diff
@@ -0,0 +1,77 @@
+package boltdb
+
+import (
+	"fmt"
+	"strings"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+const (
+	// topBucket is the bucket at the root level that contains all data related
+	// to projection OCC.
+	topBucket = "projection_occ"
+)
+
+// mkBucketAll creates bucket hierarchy and returns the deepest bucket in it.
+// Slice bb represents the bucket hierarchy with the first element being the top
+// level bucket and the last being the bucket nested at len(bb) level.
+//
+// In order to successfully create a bucket hierarchy, tx must be writable or an
+// error is returned. If some or all buckets in bb already exist, this function
+// ignores creation.
+func mkBucketAll(tx *bolt.Tx, bb ...string) (*bolt.Bucket, error) {
```

Comment:
Since the first parameter to `mkBucketAll()` is *always* `topBucket` and we're only ever getting a "handler bucket", I think it would be better to refactor this and `bucket()` to show that. It avoids the need to explain `bb`, and the need for the anonymous interfaces.

```suggestion
func makeBucket(tx *bolt.Tx, hk string) (*bolt.Bucket, error) {
```

Reply:
May be it's worth renaming `makeBucket()` and `bucket()` functions to `makeHandlerBkt()` and `getHandlerBkt()` respectively? Additionally, would we rephrase all panic messages and comments to make it sound more specific about handler-key/bucket?

Reply:
have been fixed in the latest commits.